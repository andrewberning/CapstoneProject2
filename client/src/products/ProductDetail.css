.ProductDetail {
  margin:  20px 50px;
  font-weight: bold;
}

.ProductDetail .product-container {
  width: 100%;
  height: 400px;
  display: flex;
  border: 2px solid rgba(0, 0, 0, 0.175);
  border-radius: 15px;
}

.ProductDetail .product-img-container {
  width: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.ProductDetail .product-img-container .product-img {
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
}

.ProductDetail .product-details-container {
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 75px;
}