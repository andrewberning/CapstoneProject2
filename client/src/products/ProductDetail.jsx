import { useState, useEffect, useContext } from "react";
import { useParams } from "react-router-dom";
import ShoplyApi from "../api/api";
import UserContext from "../auth/UserContext";


export default function ProductDetail() {
  const { id, category } = useParams();
  const { currentUser, hasAddedToCart, addToCart } = useContext(UserContext);
  const [product, setProduct] = useState(null);

  useEffect(function getProductForUser() {
    async function getProduct() {
      setProduct(await ShoplyApi.getProduct(id, category));
    }

    getProduct();
  }, [id, category]);

    
  async function handleAddToCart(evt) {
    evt.preventDefault();
    // if (added) return;
    addToCart(currentUser, product , 1);
    // setAdded(true);
  }

  if (!product) return (
    <h2>Product not found.</h2>
  );

  return (
    
    <div className="ProductDetail card">
      <div className="product-img-container w-25">
        <img src={product.image_url} className="card-img" />
      </div>
      <div className="card-body">
        <div className="cart-item-details">
          <div className="card-title">{product.name}</div>
          <div className="card-text">{product.description}</div>
          <div className="card-price">${product.price}</div>
        </div>
      </div>
      <button onClick={handleAddToCart} className="btn btn-primary mt-2">
        Add to Cart
      </button>
    </div>
  )
}