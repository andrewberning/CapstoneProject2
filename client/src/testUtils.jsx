import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
