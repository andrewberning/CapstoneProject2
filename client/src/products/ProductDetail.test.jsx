import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ProductDetail from "./ProductDetail";
import { UserProvider } from "../testUtils";
// import ShoplyApi from "../api/api";

// Mock the ShoplyApi module
vi.mock("../api/api", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    getProduct: vi.fn().mockResolvedValue({
      id: '1',
      name: 'Sample Product',
      description: 'This is a sample product.',
      price: 19.99,
      image_url: 'http://example.com/sample-product.jpg',
    }),
  };
});

describe('ProductDetail component', () => {

  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <ProductDetail />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <ProductDetail />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
})