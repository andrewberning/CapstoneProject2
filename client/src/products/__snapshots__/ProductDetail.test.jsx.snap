// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductDetail component > matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Product not found.
  </h2>
</DocumentFragment>
`;
